src/vellum/workflows/nodes/mocks.py

```diff
@@ -13,5 +105,137 @@ class MockNodeExecution(UniversalBaseModel):
+                return lhs.equals(rhs)
+            elif raw_logical_condition.operator == "!=":
+                return lhs.does_not_equal(rhs)
+            else:
+                raise WorkflowInitializationException(f"Unsupported logical operator: {raw_logical_condition.operator}")
+
+        def _translate_raw_logical_expression_variable(
+            raw_variable: _RawMockWorkflowNodeExecutionValuePointer,
+        ) -> BaseDescriptor:
+            if raw_variable.type == "CONSTANT_VALUE":
+                return ConstantValueReference(raw_variable.variable_value.value)
+            elif raw_variable.type == "EXECUTION_COUNTER":
+                node = nodes[raw_variable.node_id]
+                return node.Execution.count
+            else:
+                raise WorkflowInitializationException(f"Unsupported logical expression type: {raw_variable.type}")
+
+        mock_node_executions = []
+        for mock_workflow_node_config in mock_workflow_node_configs:
+            for mock_execution in mock_workflow_node_config.mock_executions:
+                try:
+                    when_condition = _translate_raw_logical_expression(
+                        mock_execution.when_condition.expression,
+                        mock_execution.when_condition.variables,
+                    )
+
+                    then_outputs = nodes[mock_workflow_node_config.node_id].Outputs()
+                    for then_output in mock_execution.then_outputs:
+                        node_output_name = node_output_name_by_id.get(then_output.output_id)
+                        if node_output_name is None:
+                            raise WorkflowInitializationException(
+                                f"Output {then_output.output_id} not found in node {mock_workflow_node_config.node_id}"
+                            )
+
+                        resolved_output_reference = _translate_raw_logical_expression_variable(then_output.value)
+                        if isinstance(resolved_output_reference, ConstantValueReference):
+                            setattr(
+                                then_outputs,
+                                node_output_name,
+                                resolved_output_reference._value,
+                            )
+                        else:
+                            raise WorkflowInitializationException(
+                                f"Unsupported resolved output reference type: {type(resolved_output_reference)}"
+                            )
+
+                    mock_node_executions.append(
+                        MockNodeExecution(
+                            when_condition=when_condition,
+                            then_outputs=then_outputs,
+                        )
+                    )
+                except Exception as e:
+                    logger.exception("Failed to validate mock node execution", exc_info=e)
+                    continue
```

Comment:
Skipping mock node executions for now if one fails, viewing it as the lesser evil than the workflow failing to initialize